epic_kitchens/dataset/epic_dataset.py

```diff
@@ -90,17 +90,45 @@ def __repr__(self):
 class EpicVideoDataset(VideoDataset):
 
     def __init__(
-        self, gulp_path: Path, class_type: str, class_getter=None, with_metadata=False
+        self,
+        gulp_path: Path,
+        class_type: str,
+        *,
+        with_metadata: bool = False,
+        class_getter: Optional[Callable[[Dict[str, Any]], Any]] = None,
+        segment_filter: Optional[Callable[[VideoSegment], bool]] = None,
+        sample_transform: Optional[Callable[[List[PIL.Image.Image]], List[PIL.Image.Image]]] = None
     ) -> None:
-        super().__init__(_class_count[class_type])
+        """
+
+        Parameters
+        ----------
+        gulp_path
+            Path to gulp directory containing the gulped EPIC RGB or flow frames
+        class_type
+            One of verb, noun, verb+noun, determines what label the segment returns
+        class_getter
+            Optionally provide a callable that takes in the gulp dict representing the segment from which
+            you should return the class you wish the segment to have
+        with_metadata
+            When True the segments will yield a tuple (metadata, class) where the class is defined by the
+            class getter and the metadata is the raw dictionary stored in the gulp file.
+        segment_filter
+            Optionally provide a callable that takes a segment and returns True if you want to keep the
+            segment in the dataset, or False if you wish to exclude it.
+        sample_filter
+            Optionally provide a sample transform function which takes a list of PIL images and transforms
+            each of them. This is applied on the list of frames upon loading the dataset.
```

Comment:
"This is applied on the frames just before returning from `load_frames`"

Reply:
Fixed in 52af03f